hips/hip-0012.md

```diff
@@ -0,0 +1,186 @@
+and the resources of CNCF will be utilized for advertising this meeting and its purposes.
+
+During all stages of Helm 4 development, the Release Engineer will ensure that an update on progress is given at no fewer than two Helm public dev calls each month.
+
+## Rejected ideas
+
+Helm 3 was largely developed along an informal version of the above model.
+However, we were criticized for not documenting the process or explaining how we were going to follow the process.
+This proposal provides an alternative to an "informal process" that would draw similar ire.
+
+Multiple release managers was also considered.
+However, two things suggest multiple release managers would not be appropriate:
+
+1. The desire to have a single authority for architectural integrity and resolution
+2. The simple fact that there are not enough Helm core maintainers to justify having a committee
+
+We considered allowing a non-core maintainer be a release engineer, but rejected for the following reasons:
+
+1. A release engineer needs detailed information about Helm's design, intents, and implementation. Our only way to ensure this is via maintainer-ship.
+2. A non-core maintainer is not bound by the core maintainer process, and this would introduce tension and exceptionalism.
+
+It has been suggested that Microsoft should lead the development process, as it was MS engineers who started Helm, and MS engineers are the most frequent contributors to Helm.
+We have decided there is no reason why any other core maintainer should be ruled out in virtue of their chosen employment.
+We also believe such a restriction runs contrary to CNCF's intentions.
+
+We considered allowing any Helm maintainer from any project to be the release engineer for Helm.
+However, we determined that the special knowledge of the Helm core codebase is requisite for successfully fulfilling this role, especially as pertains to the technical decision making required in this role. We consider the release engineer to be more of an architect's role than a project manager's.
+
+We considered restricting release engineer to just org maintainers. However, it is a stated goal of the org maintainers that they be focused on the organizational (not technical) aspects of the project.
+Release engineering is beyond their charter.
+
+We considered not requiring HIPs for features and changes during the Helm 4 process.
+In the past, we have done without.
+But also, in the past we have had substantial pushback from the community that major changes
+were not discussed and were pushed through.
+Furthermore, there is an almost complete absence of documentation for why decisions were made in Helm 2.
+And in Helm 3, the documentation is limited to only a few draft documents.
+Therefore, HIPs will provide a framework for discussion as well as a documented decision making process.
+While some may complain that this is undue overhead, we believe the benefits far outweigh the burden.
```

Comment:
If this does become a visible overhead, we may want to revisit this. However, this is a great start.